File: C/2020/56限/func4.c
//  func4.c
//  cl  func4.c
#include <stdio.h>

void add(int *i, int *j) {
    (*i) = (*i) + (*j);
}

int main() {
    int i = 10;
    int j = 20;
    add(&i, &j);
    printf("%d \n", i);

    return 0;
}

File: PhpBeginner/2022/htdocs(Tue.)/func4.php
<?php  // func4.php

function func(int $num) {
    echo "in func <br>";
    var_dump($num);
    $num *= 999;
    var_dump($num);
    echo "out func <br>";
}
//
$num = 111;
func($num);
var_dump($num);

//
echo "<br><br>";
function func2(int &$num) {
    echo "<br>in func2 <br>";
    var_dump($num);
    $num *= 999;
    var_dump($num);
    echo "out func2 <br>";
}
//
$num = 111;
var_dump($num);
func2($num);
var_dump($num);








